// m3o/image/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ImageService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var ImageService = /** @class */ (function () {
    function ImageService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Convert an image from one format (jpeg, png etc.) to an other either on the fly (from base64 to base64),
    // or by uploading the conversion result.
    // To use the file parameter you need to send the request as a multipart/form-data rather than the usual application/json
    // with each parameter as a form field.
    ImageService.prototype.convert = function (request) {
        return this.client.call("image", "Convert", request);
    };
    // Delete an image previously uploaded.
    ImageService.prototype.delete = function (request) {
        return this.client.call("image", "Delete", request);
    };
    // Resize an image on the fly without storing it (by sending and receiving a base64 encoded image), or resize and upload depending on parameters.
    // If one of width or height is 0, the image aspect ratio is preserved.
    // Optional cropping.
    // To use the file parameter you need to send the request as a multipart/form-data rather than the usual application/json
    // with each parameter as a form field.
    ImageService.prototype.resize = function (request) {
        return this.client.call("image", "Resize", request);
    };
    // Upload an image by either sending a base64 encoded image to this endpoint or a URL.
    // To resize an image before uploading, see the Resize endpoint.
    // To use the file parameter you need to send the request as a multipart/form-data rather than the usual application/json
    // with each parameter as a form field.
    ImageService.prototype.upload = function (request) {
        return this.client.call("image", "Upload", request);
    };
    return ImageService;
}());
exports.ImageService = ImageService;
