// m3o/analytics/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnalyticsService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var AnalyticsService = /** @class */ (function () {
    function AnalyticsService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Delete an event
    AnalyticsService.prototype.delete = function (request) {
        return this.client.call("analytics", "Delete", request);
    };
    // List all events
    AnalyticsService.prototype.list = function (request) {
        return this.client.call("analytics", "List", request);
    };
    // Get a single event
    AnalyticsService.prototype.read = function (request) {
        return this.client.call("analytics", "Read", request);
    };
    // Track an event, it will be created if it doesn't exist
    AnalyticsService.prototype.track = function (request) {
        return this.client.call("analytics", "Track", request);
    };
    return AnalyticsService;
}());
exports.AnalyticsService = AnalyticsService;
