// m3o/db/esm/index.js
import * as m3o from "@m3o/m3o-node";
export class DbService {
    constructor(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Count records in a table
    count(request) {
        return this.client.call("db", "Count", request);
    }
    // Create a record in the database. Optionally include an "id" field otherwise it's set automatically.
    create(request) {
        return this.client.call("db", "Create", request);
    }
    // Delete a record in the database by id.
    delete(request) {
        return this.client.call("db", "Delete", request);
    }
    // Drop a table in the DB
    dropTable(request) {
        return this.client.call("db", "DropTable", request);
    }
    // List tables in the DB
    listTables(request) {
        return this.client.call("db", "ListTables", request);
    }
    // Read data from a table. Lookup can be by ID or via querying any field in the record.
    read(request) {
        return this.client.call("db", "Read", request);
    }
    // Rename a table
    renameTable(request) {
        return this.client.call("db", "RenameTable", request);
    }
    // Truncate the records in a table
    truncate(request) {
        return this.client.call("db", "Truncate", request);
    }
    // Update a record in the database. Include an "id" in the record to update.
    update(request) {
        return this.client.call("db", "Update", request);
    }
}
