// m3o/app/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var AppService = /** @class */ (function () {
    function AppService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Delete an app
    AppService.prototype.delete = function (request) {
        return this.client.call("app", "Delete", request);
    };
    // List all the apps
    AppService.prototype.list = function (request) {
        return this.client.call("app", "List", request);
    };
    // Get the logs for an app
    AppService.prototype.logs = function (request) {
        return this.client.call("app", "Logs", request);
    };
    // Return the support regions
    AppService.prototype.regions = function (request) {
        return this.client.call("app", "Regions", request);
    };
    // Reserve app names
    AppService.prototype.reserve = function (request) {
        return this.client.call("app", "Reserve", request);
    };
    // Resolve an app by id to its raw backend endpoint
    AppService.prototype.resolve = function (request) {
        return this.client.call("app", "Resolve", request);
    };
    // Run an app from source
    AppService.prototype.run = function (request) {
        return this.client.call("app", "Run", request);
    };
    // Get the status of an app
    AppService.prototype.status = function (request) {
        return this.client.call("app", "Status", request);
    };
    // Update the app. The latest source code will be downloaded, built and deployed.
    AppService.prototype.update = function (request) {
        return this.client.call("app", "Update", request);
    };
    return AppService;
}());
exports.AppService = AppService;
