// m3o/nft/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.NftService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var NftService = /** @class */ (function () {
    function NftService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Get a single asset by the contract
    NftService.prototype.asset = function (request) {
        return this.client.call("nft", "Asset", request);
    };
    // Return a list of assets
    NftService.prototype.assets = function (request) {
        return this.client.call("nft", "Assets", request);
    };
    // Get a collection by its slug
    NftService.prototype.collection = function (request) {
        return this.client.call("nft", "Collection", request);
    };
    // Get a list of collections
    NftService.prototype.collections = function (request) {
        return this.client.call("nft", "Collections", request);
    };
    // Create your own NFT (coming soon)
    NftService.prototype.create = function (request) {
        return this.client.call("nft", "Create", request);
    };
    return NftService;
}());
exports.NftService = NftService;
