// m3o/quran/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.QuranService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var QuranService = /** @class */ (function () {
    function QuranService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // List the Chapters (surahs) of the Quran
    QuranService.prototype.chapters = function (request) {
        return this.client.call("quran", "Chapters", request);
    };
    // Search the Quran for any form of query or questions
    QuranService.prototype.search = function (request) {
        return this.client.call("quran", "Search", request);
    };
    // Get a summary for a given chapter (surah)
    QuranService.prototype.summary = function (request) {
        return this.client.call("quran", "Summary", request);
    };
    // Lookup the verses (ayahs) for a chapter including
    // translation, interpretation and breakdown by individual
    // words.
    QuranService.prototype.verses = function (request) {
        return this.client.call("quran", "Verses", request);
    };
    return QuranService;
}());
exports.QuranService = QuranService;
