// m3o/cache/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CacheService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var CacheService = /** @class */ (function () {
    function CacheService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Decrement a value (if it's a number). If key not found it is equivalent to set.
    CacheService.prototype.decrement = function (request) {
        return this.client.call("cache", "Decrement", request);
    };
    // Delete a value from the cache. If key not found a success response is returned.
    CacheService.prototype.delete = function (request) {
        return this.client.call("cache", "Delete", request);
    };
    // Get an item from the cache by key. If key is not found, an empty response is returned.
    CacheService.prototype.get = function (request) {
        return this.client.call("cache", "Get", request);
    };
    // Increment a value (if it's a number). If key not found it is equivalent to set.
    CacheService.prototype.increment = function (request) {
        return this.client.call("cache", "Increment", request);
    };
    // List all the available keys
    CacheService.prototype.listKeys = function (request) {
        return this.client.call("cache", "ListKeys", request);
    };
    // Set an item in the cache. Overwrites any existing value already set.
    CacheService.prototype.set = function (request) {
        return this.client.call("cache", "Set", request);
    };
    return CacheService;
}());
exports.CacheService = CacheService;
