// m3o/url/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UrlService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var UrlService = /** @class */ (function () {
    function UrlService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Create a URL
    UrlService.prototype.create = function (request) {
        return this.client.call("url", "Create", request);
    };
    // Delete a URL
    UrlService.prototype.delete = function (request) {
        return this.client.call("url", "Delete", request);
    };
    // List all the shortened URLs
    UrlService.prototype.list = function (request) {
        return this.client.call("url", "List", request);
    };
    // Resolve returns the destination URL of a short URL.
    UrlService.prototype.resolve = function (request) {
        return this.client.call("url", "Resolve", request);
    };
    // Shorten a URL
    UrlService.prototype.shorten = function (request) {
        return this.client.call("url", "Shorten", request);
    };
    // Update the destination for a short URL
    UrlService.prototype.update = function (request) {
        return this.client.call("url", "Update", request);
    };
    return UrlService;
}());
exports.UrlService = UrlService;
