// m3o/wallet/esm/index.js
import * as m3o from "@m3o/m3o-node";
export class WalletService {
    constructor(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Get the balance of a wallet
    balance(request) {
        return this.client.call("wallet", "Balance", request);
    }
    // Create a new wallet
    create(request) {
        return this.client.call("wallet", "Create", request);
    }
    // Add credit to a wallet
    credit(request) {
        return this.client.call("wallet", "Credit", request);
    }
    // Debit a wallet
    debit(request) {
        return this.client.call("wallet", "Debit", request);
    }
    // Delete a wallet
    delete(request) {
        return this.client.call("wallet", "Delete", request);
    }
    // List your wallets
    list(request) {
        return this.client.call("wallet", "List", request);
    }
    // Get wallet by id
    read(request) {
        return this.client.call("wallet", "Read", request);
    }
    // List the transactions for a wallet
    transactions(request) {
        return this.client.call("wallet", "Transactions", request);
    }
    // Make a transfer from one wallet to another
    transfer(request) {
        return this.client.call("wallet", "Transfer", request);
    }
}
