// m3o/file/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var FileService = /** @class */ (function () {
    function FileService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Delete a file by project name/path
    FileService.prototype.delete = function (request) {
        return this.client.call("file", "Delete", request);
    };
    // List files by their project and optionally a path.
    FileService.prototype.list = function (request) {
        return this.client.call("file", "List", request);
    };
    // Read a file by path
    FileService.prototype.read = function (request) {
        return this.client.call("file", "Read", request);
    };
    // Save a file
    FileService.prototype.save = function (request) {
        return this.client.call("file", "Save", request);
    };
    return FileService;
}());
exports.FileService = FileService;
