// m3o/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
var address = __importStar(require("./address"));
var ai = __importStar(require("./ai"));
var analytics = __importStar(require("./analytics"));
var answer = __importStar(require("./answer"));
var app = __importStar(require("./app"));
var avatar = __importStar(require("./avatar"));
var bitcoin = __importStar(require("./bitcoin"));
var cache = __importStar(require("./cache"));
var carbon = __importStar(require("./carbon"));
var chat = __importStar(require("./chat"));
var comments = __importStar(require("./comments"));
var contact = __importStar(require("./contact"));
var cron = __importStar(require("./cron"));
var crypto = __importStar(require("./crypto"));
var currency = __importStar(require("./currency"));
var db = __importStar(require("./db"));
var dns = __importStar(require("./dns"));
var email = __importStar(require("./email"));
var emoji = __importStar(require("./emoji"));
var ethereum = __importStar(require("./ethereum"));
var evchargers = __importStar(require("./evchargers"));
var event = __importStar(require("./event"));
var file = __importStar(require("./file"));
var forex = __importStar(require("./forex"));
var fx = __importStar(require("./function"));
var geocoding = __importStar(require("./geocoding"));
var gifs = __importStar(require("./gifs"));
var google = __importStar(require("./google"));
var helloworld = __importStar(require("./helloworld"));
var holidays = __importStar(require("./holidays"));
var id = __importStar(require("./id"));
var image = __importStar(require("./image"));
var ip = __importStar(require("./ip"));
var joke = __importStar(require("./joke"));
var lists = __importStar(require("./lists"));
var location = __importStar(require("./location"));
var memegen = __importStar(require("./memegen"));
var minecraft = __importStar(require("./minecraft"));
var movie = __importStar(require("./movie"));
var mq = __importStar(require("./mq"));
var news = __importStar(require("./news"));
var nft = __importStar(require("./nft"));
var notes = __importStar(require("./notes"));
var otp = __importStar(require("./otp"));
var password = __importStar(require("./password"));
var ping = __importStar(require("./ping"));
var place = __importStar(require("./place"));
var postcode = __importStar(require("./postcode"));
var prayer = __importStar(require("./prayer"));
var price = __importStar(require("./price"));
var qr = __importStar(require("./qr"));
var quran = __importStar(require("./quran"));
var routing = __importStar(require("./routing"));
var rss = __importStar(require("./rss"));
var search = __importStar(require("./search"));
var secret = __importStar(require("./secret"));
var sentiment = __importStar(require("./sentiment"));
var sms = __importStar(require("./sms"));
var space = __importStar(require("./space"));
var spam = __importStar(require("./spam"));
var stock = __importStar(require("./stock"));
var stream = __importStar(require("./stream"));
var sunnah = __importStar(require("./sunnah"));
var thumbnail = __importStar(require("./thumbnail"));
var time = __importStar(require("./time"));
var translate = __importStar(require("./translate"));
var tunnel = __importStar(require("./tunnel"));
var twitter = __importStar(require("./twitter"));
var url = __importStar(require("./url"));
var user = __importStar(require("./user"));
var vehicle = __importStar(require("./vehicle"));
var wallet = __importStar(require("./wallet"));
var weather = __importStar(require("./weather"));
var wordle = __importStar(require("./wordle"));
var youtube = __importStar(require("./youtube"));
var Client = /** @class */ (function () {
    function Client(token) {
        this.address = new address.AddressService(token);
        this.ai = new ai.AiService(token);
        this.analytics = new analytics.AnalyticsService(token);
        this.answer = new answer.AnswerService(token);
        this.app = new app.AppService(token);
        this.avatar = new avatar.AvatarService(token);
        this.bitcoin = new bitcoin.BitcoinService(token);
        this.cache = new cache.CacheService(token);
        this.carbon = new carbon.CarbonService(token);
        this.chat = new chat.ChatService(token);
        this.comments = new comments.CommentsService(token);
        this.contact = new contact.ContactService(token);
        this.cron = new cron.CronService(token);
        this.crypto = new crypto.CryptoService(token);
        this.currency = new currency.CurrencyService(token);
        this.db = new db.DbService(token);
        this.dns = new dns.DnsService(token);
        this.email = new email.EmailService(token);
        this.emoji = new emoji.EmojiService(token);
        this.ethereum = new ethereum.EthereumService(token);
        this.evchargers = new evchargers.EvchargersService(token);
        this.event = new event.EventService(token);
        this.file = new file.FileService(token);
        this.forex = new forex.ForexService(token);
        this.function = new fx.FunctionService(token);
        this.geocoding = new geocoding.GeocodingService(token);
        this.gifs = new gifs.GifsService(token);
        this.google = new google.GoogleService(token);
        this.helloworld = new helloworld.HelloworldService(token);
        this.holidays = new holidays.HolidaysService(token);
        this.id = new id.IdService(token);
        this.image = new image.ImageService(token);
        this.ip = new ip.IpService(token);
        this.joke = new joke.JokeService(token);
        this.lists = new lists.ListsService(token);
        this.location = new location.LocationService(token);
        this.memegen = new memegen.MemegenService(token);
        this.minecraft = new minecraft.MinecraftService(token);
        this.movie = new movie.MovieService(token);
        this.mq = new mq.MqService(token);
        this.news = new news.NewsService(token);
        this.nft = new nft.NftService(token);
        this.notes = new notes.NotesService(token);
        this.otp = new otp.OtpService(token);
        this.password = new password.PasswordService(token);
        this.ping = new ping.PingService(token);
        this.place = new place.PlaceService(token);
        this.postcode = new postcode.PostcodeService(token);
        this.prayer = new prayer.PrayerService(token);
        this.price = new price.PriceService(token);
        this.qr = new qr.QrService(token);
        this.quran = new quran.QuranService(token);
        this.routing = new routing.RoutingService(token);
        this.rss = new rss.RssService(token);
        this.search = new search.SearchService(token);
        this.secret = new secret.SecretService(token);
        this.sentiment = new sentiment.SentimentService(token);
        this.sms = new sms.SmsService(token);
        this.space = new space.SpaceService(token);
        this.spam = new spam.SpamService(token);
        this.stock = new stock.StockService(token);
        this.stream = new stream.StreamService(token);
        this.sunnah = new sunnah.SunnahService(token);
        this.thumbnail = new thumbnail.ThumbnailService(token);
        this.time = new time.TimeService(token);
        this.translate = new translate.TranslateService(token);
        this.tunnel = new tunnel.TunnelService(token);
        this.twitter = new twitter.TwitterService(token);
        this.url = new url.UrlService(token);
        this.user = new user.UserService(token);
        this.vehicle = new vehicle.VehicleService(token);
        this.wallet = new wallet.WalletService(token);
        this.weather = new weather.WeatherService(token);
        this.wordle = new wordle.WordleService(token);
        this.youtube = new youtube.YoutubeService(token);
    }
    return Client;
}());
exports.Client = Client;
exports.default = (function (token) {
    if (token === void 0) { token = process.env.M3O_API_TOKEN; }
    return {
        address: new address.AddressService(token),
        ai: new ai.AiService(token),
        analytics: new analytics.AnalyticsService(token),
        answer: new answer.AnswerService(token),
        app: new app.AppService(token),
        avatar: new avatar.AvatarService(token),
        bitcoin: new bitcoin.BitcoinService(token),
        cache: new cache.CacheService(token),
        carbon: new carbon.CarbonService(token),
        chat: new chat.ChatService(token),
        comments: new comments.CommentsService(token),
        contact: new contact.ContactService(token),
        cron: new cron.CronService(token),
        crypto: new crypto.CryptoService(token),
        currency: new currency.CurrencyService(token),
        db: new db.DbService(token),
        dns: new dns.DnsService(token),
        email: new email.EmailService(token),
        emoji: new emoji.EmojiService(token),
        ethereum: new ethereum.EthereumService(token),
        evchargers: new evchargers.EvchargersService(token),
        event: new event.EventService(token),
        file: new file.FileService(token),
        forex: new forex.ForexService(token),
        function: new fx.FunctionService(token),
        geocoding: new geocoding.GeocodingService(token),
        gifs: new gifs.GifsService(token),
        google: new google.GoogleService(token),
        helloworld: new helloworld.HelloworldService(token),
        holidays: new holidays.HolidaysService(token),
        id: new id.IdService(token),
        image: new image.ImageService(token),
        ip: new ip.IpService(token),
        joke: new joke.JokeService(token),
        lists: new lists.ListsService(token),
        location: new location.LocationService(token),
        memegen: new memegen.MemegenService(token),
        minecraft: new minecraft.MinecraftService(token),
        movie: new movie.MovieService(token),
        mq: new mq.MqService(token),
        news: new news.NewsService(token),
        nft: new nft.NftService(token),
        notes: new notes.NotesService(token),
        otp: new otp.OtpService(token),
        password: new password.PasswordService(token),
        ping: new ping.PingService(token),
        place: new place.PlaceService(token),
        postcode: new postcode.PostcodeService(token),
        prayer: new prayer.PrayerService(token),
        price: new price.PriceService(token),
        qr: new qr.QrService(token),
        quran: new quran.QuranService(token),
        routing: new routing.RoutingService(token),
        rss: new rss.RssService(token),
        search: new search.SearchService(token),
        secret: new secret.SecretService(token),
        sentiment: new sentiment.SentimentService(token),
        sms: new sms.SmsService(token),
        space: new space.SpaceService(token),
        spam: new spam.SpamService(token),
        stock: new stock.StockService(token),
        stream: new stream.StreamService(token),
        sunnah: new sunnah.SunnahService(token),
        thumbnail: new thumbnail.ThumbnailService(token),
        time: new time.TimeService(token),
        translate: new translate.TranslateService(token),
        tunnel: new tunnel.TunnelService(token),
        twitter: new twitter.TwitterService(token),
        url: new url.UrlService(token),
        user: new user.UserService(token),
        vehicle: new vehicle.VehicleService(token),
        wallet: new wallet.WalletService(token),
        weather: new weather.WeatherService(token),
        wordle: new wordle.WordleService(token),
        youtube: new youtube.YoutubeService(token),
    };
});
