// m3o/twitter/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.TwitterService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var TwitterService = /** @class */ (function () {
    function TwitterService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Search for tweets with a simple query
    TwitterService.prototype.search = function (request) {
        return this.client.call("twitter", "Search", request);
    };
    // Get the timeline for a given user
    TwitterService.prototype.timeline = function (request) {
        return this.client.call("twitter", "Timeline", request);
    };
    // Get the current global trending topics
    TwitterService.prototype.trends = function (request) {
        return this.client.call("twitter", "Trends", request);
    };
    // Get a user's twitter profile
    TwitterService.prototype.user = function (request) {
        return this.client.call("twitter", "User", request);
    };
    return TwitterService;
}());
exports.TwitterService = TwitterService;
