// m3o/contact/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ContactService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var ContactService = /** @class */ (function () {
    function ContactService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Create a contact
    ContactService.prototype.create = function (request) {
        return this.client.call("contact", "Create", request);
    };
    // Delete a contact
    ContactService.prototype.delete = function (request) {
        return this.client.call("contact", "Delete", request);
    };
    // List contacts
    ContactService.prototype.list = function (request) {
        return this.client.call("contact", "List", request);
    };
    // Read contact details
    ContactService.prototype.read = function (request) {
        return this.client.call("contact", "Read", request);
    };
    // Update a contact
    ContactService.prototype.update = function (request) {
        return this.client.call("contact", "Update", request);
    };
    return ContactService;
}());
exports.ContactService = ContactService;
