// m3o/search/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var SearchService = /** @class */ (function () {
    function SearchService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Create an index by name
    SearchService.prototype.createIndex = function (request) {
        return this.client.call("search", "CreateIndex", request);
    };
    // Delete an index by name
    SearchService.prototype.deleteIndex = function (request) {
        return this.client.call("search", "DeleteIndex", request);
    };
    // Delete a record given its ID
    SearchService.prototype.delete = function (request) {
        return this.client.call("search", "Delete", request);
    };
    // Index a record i.e. insert a document to search for.
    SearchService.prototype.index = function (request) {
        return this.client.call("search", "Index", request);
    };
    // Search for records in a given in index
    SearchService.prototype.search = function (request) {
        return this.client.call("search", "Search", request);
    };
    return SearchService;
}());
exports.SearchService = SearchService;
