// m3o/rss/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RssService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var RssService = /** @class */ (function () {
    function RssService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Add a new RSS feed with a name, url, and category
    RssService.prototype.add = function (request) {
        return this.client.call("rss", "Add", request);
    };
    // Get an RSS feed by name. If no name is given, all feeds are returned. Default limit is 25 entries.
    RssService.prototype.feed = function (request) {
        return this.client.call("rss", "Feed", request);
    };
    // List the saved RSS fields
    RssService.prototype.list = function (request) {
        return this.client.call("rss", "List", request);
    };
    // Remove an RSS feed by name
    RssService.prototype.remove = function (request) {
        return this.client.call("rss", "Remove", request);
    };
    return RssService;
}());
exports.RssService = RssService;
