// m3o/comments/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CommentsService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var CommentsService = /** @class */ (function () {
    function CommentsService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Create a new comment
    CommentsService.prototype.create = function (request) {
        return this.client.call("comments", "Create", request);
    };
    // Delete a comment
    CommentsService.prototype.delete = function (request) {
        return this.client.call("comments", "Delete", request);
    };
    // Subscribe to comments events
    CommentsService.prototype.events = function (request) {
        return this.client.stream("comments", "Events", request);
    };
    // List all the comments
    CommentsService.prototype.list = function (request) {
        return this.client.call("comments", "List", request);
    };
    // Read a comment
    CommentsService.prototype.read = function (request) {
        return this.client.call("comments", "Read", request);
    };
    // Update a comment
    CommentsService.prototype.update = function (request) {
        return this.client.call("comments", "Update", request);
    };
    return CommentsService;
}());
exports.CommentsService = CommentsService;
