// m3o/function/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var FunctionService = /** @class */ (function () {
    function FunctionService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Call a function by name
    FunctionService.prototype.call = function (request) {
        return this.client.call("function", "Call", request);
    };
    // Delete a function by name
    FunctionService.prototype.delete = function (request) {
        return this.client.call("function", "Delete", request);
    };
    // Deploy a group of functions
    FunctionService.prototype.deploy = function (request) {
        return this.client.call("function", "Deploy", request);
    };
    // Get the info for a deployed function
    FunctionService.prototype.describe = function (request) {
        return this.client.call("function", "Describe", request);
    };
    // List all the deployed functions
    FunctionService.prototype.list = function (request) {
        return this.client.call("function", "List", request);
    };
    // Get the logs for a function
    FunctionService.prototype.logs = function (request) {
        return this.client.call("function", "Logs", request);
    };
    // Return the backend url for proxying
    FunctionService.prototype.proxy = function (request) {
        return this.client.call("function", "Proxy", request);
    };
    // Return a list of supported regions
    FunctionService.prototype.regions = function (request) {
        return this.client.call("function", "Regions", request);
    };
    // Reserve function names and resources beyond free quota
    FunctionService.prototype.reserve = function (request) {
        return this.client.call("function", "Reserve", request);
    };
    // Return a list of supported runtimes
    FunctionService.prototype.runtimes = function (request) {
        return this.client.call("function", "Runtimes", request);
    };
    // Update a function. Downloads the source, builds and redeploys
    FunctionService.prototype.update = function (request) {
        return this.client.call("function", "Update", request);
    };
    return FunctionService;
}());
exports.FunctionService = FunctionService;
