// m3o/wallet/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var WalletService = /** @class */ (function () {
    function WalletService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Get the balance of a wallet
    WalletService.prototype.balance = function (request) {
        return this.client.call("wallet", "Balance", request);
    };
    // Create a new wallet
    WalletService.prototype.create = function (request) {
        return this.client.call("wallet", "Create", request);
    };
    // Add credit to a wallet
    WalletService.prototype.credit = function (request) {
        return this.client.call("wallet", "Credit", request);
    };
    // Debit a wallet
    WalletService.prototype.debit = function (request) {
        return this.client.call("wallet", "Debit", request);
    };
    // Delete a wallet
    WalletService.prototype.delete = function (request) {
        return this.client.call("wallet", "Delete", request);
    };
    // List your wallets
    WalletService.prototype.list = function (request) {
        return this.client.call("wallet", "List", request);
    };
    // Get wallet by id
    WalletService.prototype.read = function (request) {
        return this.client.call("wallet", "Read", request);
    };
    // List the transactions for a wallet
    WalletService.prototype.transactions = function (request) {
        return this.client.call("wallet", "Transactions", request);
    };
    // Make a transfer from one wallet to another
    WalletService.prototype.transfer = function (request) {
        return this.client.call("wallet", "Transfer", request);
    };
    return WalletService;
}());
exports.WalletService = WalletService;
