// m3o/crypto/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CryptoService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var CryptoService = /** @class */ (function () {
    function CryptoService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Returns the history for the previous close
    CryptoService.prototype.history = function (request) {
        return this.client.call("crypto", "History", request);
    };
    // Get news related to a currency
    CryptoService.prototype.news = function (request) {
        return this.client.call("crypto", "News", request);
    };
    // Get the last price for a given crypto ticker
    CryptoService.prototype.price = function (request) {
        return this.client.call("crypto", "Price", request);
    };
    // Get the last quote for a given crypto ticker
    CryptoService.prototype.quote = function (request) {
        return this.client.call("crypto", "Quote", request);
    };
    // Returns the full list of supported symbols
    CryptoService.prototype.symbols = function (request) {
        return this.client.call("crypto", "Symbols", request);
    };
    return CryptoService;
}());
exports.CryptoService = CryptoService;
