// m3o/price/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var PriceService = /** @class */ (function () {
    function PriceService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Add a price
    PriceService.prototype.add = function (request) {
        return this.client.call("price", "Add", request);
    };
    // Get the price of anything
    PriceService.prototype.get = function (request) {
        return this.client.call("price", "Get", request);
    };
    // Get the index for available prices
    PriceService.prototype.index = function (request) {
        return this.client.call("price", "Index", request);
    };
    // List prices for a given currency
    PriceService.prototype.list = function (request) {
        return this.client.call("price", "List", request);
    };
    // Report an invalid price
    PriceService.prototype.report = function (request) {
        return this.client.call("price", "Report", request);
    };
    return PriceService;
}());
exports.PriceService = PriceService;
