// m3o/lists/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListsService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var ListsService = /** @class */ (function () {
    function ListsService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Create a new list
    ListsService.prototype.create = function (request) {
        return this.client.call("lists", "Create", request);
    };
    // Delete a list
    ListsService.prototype.delete = function (request) {
        return this.client.call("lists", "Delete", request);
    };
    // Subscribe to lists events
    ListsService.prototype.events = function (request) {
        return this.client.stream("lists", "Events", request);
    };
    // List all the lists
    ListsService.prototype.list = function (request) {
        return this.client.call("lists", "List", request);
    };
    // Read a list
    ListsService.prototype.read = function (request) {
        return this.client.call("lists", "Read", request);
    };
    // Update a list
    ListsService.prototype.update = function (request) {
        return this.client.call("lists", "Update", request);
    };
    return ListsService;
}());
exports.ListsService = ListsService;
