// m3o/sunnah/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SunnahService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var SunnahService = /** @class */ (function () {
    function SunnahService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Get a list of books from within a collection. A book can contain many chapters
    // each with its own hadiths.
    SunnahService.prototype.books = function (request) {
        return this.client.call("sunnah", "Books", request);
    };
    // Get all the chapters of a given book within a collection.
    SunnahService.prototype.chapters = function (request) {
        return this.client.call("sunnah", "Chapters", request);
    };
    // Get a list of available collections. A collection is
    // a compilation of hadiths collected and written by an author.
    SunnahService.prototype.collections = function (request) {
        return this.client.call("sunnah", "Collections", request);
    };
    // Hadiths returns a list of hadiths and their corresponding text for a
    // given book within a collection.
    SunnahService.prototype.hadiths = function (request) {
        return this.client.call("sunnah", "Hadiths", request);
    };
    return SunnahService;
}());
exports.SunnahService = SunnahService;
