// m3o/stream/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var StreamService = /** @class */ (function () {
    function StreamService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Create a channel by name
    StreamService.prototype.createChannel = function (request) {
        return this.client.call("stream", "CreateChannel", request);
    };
    // List all the active channels
    StreamService.prototype.listChannels = function (request) {
        return this.client.call("stream", "ListChannels", request);
    };
    // List messages for a given channel
    StreamService.prototype.listMessages = function (request) {
        return this.client.call("stream", "ListMessages", request);
    };
    // Send a message to the stream.
    StreamService.prototype.sendMessage = function (request) {
        return this.client.call("stream", "SendMessage", request);
    };
    return StreamService;
}());
exports.StreamService = StreamService;
