// m3o/db/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DbService = void 0;
var m3o = __importStar(require("@m3o/m3o-node"));
var DbService = /** @class */ (function () {
    function DbService(token) {
        this.client = new m3o.Client({ token: token });
    }
    // Count records in a table
    DbService.prototype.count = function (request) {
        return this.client.call("db", "Count", request);
    };
    // Create a record in the database. Optionally include an "id" field otherwise it's set automatically.
    DbService.prototype.create = function (request) {
        return this.client.call("db", "Create", request);
    };
    // Delete a record in the database by id.
    DbService.prototype.delete = function (request) {
        return this.client.call("db", "Delete", request);
    };
    // Drop a table in the DB
    DbService.prototype.dropTable = function (request) {
        return this.client.call("db", "DropTable", request);
    };
    // List tables in the DB
    DbService.prototype.listTables = function (request) {
        return this.client.call("db", "ListTables", request);
    };
    // Read data from a table. Lookup can be by ID or via querying any field in the record.
    DbService.prototype.read = function (request) {
        return this.client.call("db", "Read", request);
    };
    // Rename a table
    DbService.prototype.renameTable = function (request) {
        return this.client.call("db", "RenameTable", request);
    };
    // Truncate the records in a table
    DbService.prototype.truncate = function (request) {
        return this.client.call("db", "Truncate", request);
    };
    // Update a record in the database. Include an "id" in the record to update.
    DbService.prototype.update = function (request) {
        return this.client.call("db", "Update", request);
    };
    return DbService;
}());
exports.DbService = DbService;
